v1.4 code changes

kenvynKwek opened the pull request:
- [x] bug fixes
- [x] code quality
- [x] testing improvements

kenvynKwek committed:
Update reposense authorship

Changes to features by another team member.

Contributions/authorship not recorded correctly.

Let's add reposense authorship comments to record contributions
correctly.

kenvynKwek committed:
Update for checkstyle

Wrong indentation for some lines.

Unable to pass checkstyle.

Let's update the relevant lines with the correct indentation.

kenvynKwek committed:
Update email regex

Email regex is incorrect.

Able to add invalid emails with domain last part missing.

Let's
* update the email regex to accept only valid emails.
* update relevant test cases in EmailTest.java.

kenvynKwek committed:
Update name message constraints

Name message constraints are not updated after the bugfix.

Doesn't reflect correctly when inputting an invalid name.

Let's update the name message constraints to mention only alphabets.

kenvynKwek committed:
Add test cases for 'FilterCommandTest.java'

Previous existing test cases were deleted in PR #76.

No validation on whether FilterCommand filters correctly.

Let's create test cases for 'FilterCommandTest.java'.

kenvynKwek committed:
Add more test cases for 'FilterCommandTest.java'

Insufficient test scenarios for 'FilterCommandTest.java'.

Coverage not complete.

Let's add more test cases for 'FilterCommandTest.java'.

SungMatt commented on src/main/java/seedu/address/logic/parser/FilterCommandParser.java:
```diff
@@ -41,11 +41,20 @@ public class FilterCommandParser implements Parser<FilterCommand> {
-                        PREFIX_ADDRESS, PREFIX_SALARY, PREFIX_LEAVE, PREFIX_ROLE, PREFIX_MANAGER, PREFIX_DEPARTMENT);
+                ArgumentTokenizer.tokenize(args,
+                        //@@author
+                        PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
```
Who does reposense attribute these to haha

kenvynKwek replied:
to xihua as it currently shows in reposense, instructions were to end reposense authorship comments with '//@@author' otherwise it wont know where to stop.

quite a few of the lines are the exact same as what i pushed but somehow xihua has authorship when he overwrited, i added authorship comments for what were originally my lines.

SungMatt commented on src/main/java/seedu/address/model/employee/Email.java:
```diff
@@ -27,8 +27,8 @@ public class Email {
             + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
-    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
+    private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,3}$"; // 2 to 3 characters
```
For this, we might have to follow the structure as per #88. Can't change the regex , only change the UG.

kenvynKwek replied:
noted, will revert changes.

UG updates will be in a separate PR

SungMatt commented on src/test/java/seedu/address/logic/commands/FilterCommandTest.java:
```diff
@@ -1,13 +1,369 @@
+    }
+
+    @Test
+    public void execute_noEmployeesFound() {
```
Consider renaming these to the three-part format: 

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()

as per the [Java Coding Standards](https://se-education.org/guides/conventions/java/index.html#naming)

kenvynKwek replied:
noted

kenvynKwek committed:
Revert Email regex changes

Bug considered a feature flaw.

Not under v1.4 bugfixing (out of scope).

Let's revert the changes in the email regex validation.

kenvynKwek committed:
Add test cases for 'FilterCommandParserTest.java'

Insufficient test cases for FilterCommandParser.

Insufficient coverage.

Let's add test cases for 'FilterCommandParserTest.java'.

kenvynKwek committed:
Add test cases for 'ContainsAllPredicateTest.java'

Insufficient test cases for 'ContainsAllPredicateTest.java'.

Insufficient code coverage.

Let's add more test cases for 'ContainsAllPredicateTest.java'.

kenvynKwek committed:
Add log code for 'FilterCommandParser.java'

Log info not present in filter feature.

Unable to determine user actions from logs.

Let's add log info code in 'FilterCommandParser.java'.

SungMatt commented:
Fixes #106

SungMatt commented on src/main/java/seedu/address/model/name/EmployeeName.java:
```diff
@@ -10,7 +10,7 @@
 public class EmployeeName {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
```
UG states that it can accept alphanumeric characters, as we can't change the regex.

kenvynKwek replied:
noted, will revert

SungMatt commented on src/test/java/seedu/address/model/employee/ContainsAllPredicateTest.java:
```diff
@@ -1,75 +1,181 @@
+    }
+
     @Test
     public void test_containsAddress_returnsTrue() {
```
I don't quite get the removal of these lines?

kenvynKwek replied:
because usage 'createContainsAllPredicateHelper' helper function (line 28) replaces these lines.

'predicate' is created multiple times throughout the test file and the 'createContainsAllPredicateHelper' helper function helps to create the 'predicate'

it fulfils the same purpose and the removed lines. same predicate with singular address attribute.

SungMatt commented on src/test/java/seedu/address/model/employee/ContainsAllPredicateTest.java:
```diff
@@ -1,75 +1,181 @@
     }
 
     @Test
-    public void test_differentAttributeValueBetweenPredicateAndEmployee_returnsFalse() {
```
This test too -- why was it removed?

kenvynKwek replied:
replaced by test_doesNotContainsAddress (line 113)

it tests the same thing, between amy's and bob's address -> expected to return false

SungMatt reviewed (changes requested):
The EmployeeName.java one is important to fix! The other two are less pertinent, but I'm curious as to why that is the case.

kenvynKwek committed:
Revert EmployeeName message constraints

Bug determined to be feature flaw.

Not allowed to change under v1.4, out of scope.

Let's revert the message constraint changes.
